metadata/metadata.go

```diff
@@ -511,28 +521,30 @@ func (resource *featureVariantResource) Schedule() string {
 func (resource *featureVariantResource) Dependencies(lookup ResourceLookup) (ResourceLookup, error) {
 	serialized := resource.serialized
 	depIds := []ResourceID{
-		{
-			Name:    serialized.Source.Name,
-			Variant: serialized.Source.Variant,
-			Type:    SOURCE_VARIANT,
-		},
-		{
-			Name: serialized.Entity,
-			Type: ENTITY,
-		},
 		{
 			Name: serialized.Owner,
 			Type: USER,
 		},
-		{
-			Name: serialized.Provider,
-			Type: PROVIDER,
-		},
 		{
 			Name: serialized.Name,
 			Type: FEATURE,
 		},
 	}
+	if serialized.Category == pb.FeatureVariantCategory(PRE_CALCULATED) {
```

Comment:
Feels like you should be converting the proto to match the enum , rather than converting the enum to match the proto.

Reply:
Can create a function like on the enum like `PRE_CALCULATED.equals(serialized.Category)` or something similar